test/cors-gate.spec.js

```diff
@@ -71,6 +71,21 @@ describe('cors-gate', function() {
       .expect(200, done);
   });
 
+  it('should terminate on preflight requests', function(done) {
+    this.app.use(cors({
+      origin: '*'
+    }), corsGate({
+      origin: 'http://localhost'
+    }));
+
+    this.app.post('/post', ok);
+
+    request(this.app)
+      .options('/post')
+      .set('origin', 'http://localhost:8080')
+      .expect(204, done);
```

Comment:
Could you also test the zero content length? Since that's important external behavior